_pages/documentation/contributing_beginners_guide.md

```diff
@@ -0,0 +1,315 @@
+worked on.
+
+**If, for whatever reason, you stop working on a task, please unassigned
+yourself from the task and change the task's status back to `Todo`.**
+
+## Obtaining the git repo
+
+The gem5 git repository is hosted at [](https://gem5.googlesource.com).
+**Please note: contributions made to other gem5 repos (e.g., our GitHub mirror)
+will not be considered. Please contribute to [](https://gem5.googlesource.com)
+exclusively.**
+
+To pull the gem5 git repo:
+
+```Shell
+git clone https://gem5.googlesource.com/public/gem5
+```
+
+This local repo can then be modified to incorporate the task you have assigned
+yourself.
+
+## Making modifications
+
+Different tasks will require the project to be modified in different ways.
+Though, in all cases, our style-guide must be adhered to. The full style guide
+is outlined here: [](http://gem5.org/Coding_Style).
+
+As a high-level overview:
+
+* Lines must not exceed 79 characters in length.
+* There should be no trailing white-space on any line.
+* Indentations must be 4 spaces (no tab characters).
+* Class names must use upper camel case (e.g., `ThisIsAClass`).
+* Class member variables must use lower camel case (e.g.,
+`thisIsAMemberVariable`).
+* Class member variables with their own public accessor must start with an
+underscore (e.g., `_variableWithAccessor`).
+* Local variables must use snake case (e.g., `this_is_a_local_variable`).
+* Functions must use lower camel case (e.g., `thisIsAFunction`)
+* Function parameters must use snake case.
+* Macros must be in all caps with underscores (e.g., `THIS_IS_A_MACRO`).
+* Function declaration return types must be on their own line.
+* Function brackets must be on their own line.
+* `for`/`if`/`while` branching operations must be followed by a white-space
+before the conditional statement (e.g., `for (...)`).
+* `for`/`if`/`while` branching operations' opening bracket must be on on the
+same line, with the closing bracket on its own line (e.g.,
+`for (...) {\n ... \n}\n`). There should be a space between the condition(s)
+and the opening bracket.
+
+Below is a simple toy example of how a class should be formatted:
+
+```C++
+#DEFINE EXAMPLE_MACRO 7
+class ExampleClass
+{
+    private:
+        int _fooBar;
+        int barFoo;
+
+    public:
+        int
+        getFooBar()
+        {
+            return _fooBar;
+        }
+
+        int
+        aFunction(int parameter_one, int parameter_two)
+        {
+            int local_variable = 0;
+            if (true) {
+                int local_variable = parameter_one + parameter_two + barFoo
+                                   + EXAMPLE_MACRO;
+            }
+            return local_variable;
+        }
+
+}
+```
+
+## Compiling and running tests
+
+The minimum criteria for a change to be submitted is that the code is
+compilable and the test cases pass.
+
+The following command both compiles the project and runs our system-level
```

Comment:
Do we need to require new contributors to do this?

Reply:
I think it's good practice to run the tests prior to submission. I wouldn't make this a must if our CI triggered upon submission of a new patch set, but as we need to wait for a maintainer to sign off, it may take a while for a new contributor to realize their new code simply breaks the code (or doesn't even compile). Hopefully we can loosen this restriction as we build our own CI pipelines.